gammapy/cube/tests/test_fit.py

```diff
@@ -77,75 +78,100 @@ def sky_model():
 
 
-def mask_fit(geom, sky_model):
-    p = sky_model.spatial_model.parameters
-    center = SkyCoord(p["lon_0"].value, p["lat_0"].value, frame="galactic", unit="deg")
-    circle = CircleSkyRegion(center=center, radius=1 * u.deg)
-    return geom.region_mask([circle])
+def get_map_dataset(sky_model, geom, geom_etrue, **kwargs):
+    """This computes the total npred"""
+    # define background model
+    m = Map.from_geom(geom)
+    m.quantity = 0.2 * np.ones(m.data.shape)
+    background_model = BackgroundModel(m)
 
+    psf = get_psf(geom_etrue)
+    exposure = get_exposure(geom_etrue)
 
-def counts(sky_model, exposure, background, psf, edisp):
-    """This computes the total npred"""
-    npred = MapDataset(
+    # define energy dispersion
+    e_true = geom_etrue.get_axis_by_name("energy").edges
+    e_reco = geom.get_axis_by_name("energy").edges
+    edisp = EnergyDispersion.from_diagonal_response(e_true=e_true, e_reco=e_reco)
+
+    # define fit mask
+    center = sky_model.spatial_model.position
+    circle = CircleSkyRegion(center=center, radius=1 * u.deg)
+    mask_fit = background_model.map.geom.region_mask([circle])
+
+    return MapDataset(
         model=sky_model,
         exposure=exposure,
-        background_model=background,
+        background_model=background_model,
         psf=psf,
         edisp=edisp,
-    ).npred()
-    return npred
+        mask_fit=mask_fit,
+        **kwargs
+    )
 
 
-@requires_dependency("iminuit")
 @requires_data()
-def test_map_fit(sky_model):
-    ebounds = np.logspace(-1.0, 1.0, 3)
-    ebounds_true = np.logspace(-1.0, 1.0, 4)
-    geom_r = geom(ebounds)
-    geom_t = geom_etrue(ebounds_true)
-
-    background_map = background(geom_r)
-    background_model_1 = BackgroundModel(background_map, norm=0.5)
-    background_model_2 = BackgroundModel(background_map, norm=1)
-
-    psf_map = psf(geom_t)
-    edisp_map = edisp(geom_r, geom_t)
-    exposure_map = exposure(geom_t)
-    counts_map_1 = counts(
-        sky_model, exposure_map, background_model_1, psf_map, edisp_map
-    )
-    counts_map_2 = counts(
-        sky_model, exposure_map, background_model_2, psf_map, edisp_map
-    )
+def test_map_dataset_fits_io(tmpdir, sky_model, geom, geom_etrue):
+    dataset = get_map_dataset(sky_model, geom , geom_etrue)
+    dataset.counts = dataset.npred()
+    dataset.mask_safe = dataset.mask_fit
+
+    hdulist = dataset.to_hdulist()
+
+    assert "COUNTS" in hdulist
+    assert "COUNTS_BANDS" in hdulist
+    assert "EXPOSURE" in hdulist
+    assert "EXPOSURE_BANDS" in hdulist
+    assert "BACKGROUND" in hdulist
+    assert "BACKGROUND_BANDS" in hdulist
+    assert "PSF_KERNEL" in hdulist
+    assert "PSF_KERNEL_BANDS" in hdulist
+    assert "EDISP_MATRIX" in hdulist
+    assert "EDISP_MATRIX_EBOUNDS" in hdulist
+    assert "MASK_FIT" in hdulist
+    assert "MASK_FIT_BANDS" in hdulist
+    assert "MASK_SAFE" in hdulist
+    assert "MASK_SAFE_BANDS" in hdulist
```

Comment:
There could be other stuff in `hdulist` (like e.g. extra "garbage" PrimaryHDU objects) and we won't notice. Suggest to put `actual = [_.name for _ in hdulist]` and `expected = ["COUNTS", ...]` and then `assert actual == expected`.

Reply:
Done